### PR TITLE
feat(RHTAPREL-717): use new PipelineRun builder 

### PR DESCRIPTION
**Pull request by davidmogar**
This commit uses the new builder to generate the managed PipelineRun.

Signed-off-by: David Moreno García <damoreno@redhat.com>

Depends on #310, #311 and #312

### CODE REVIEW

File: tekton/utils/pipeline.go
Comment (davidmogar) [reply tekton/utils/pipeline.go:57]: Absolutely. I'll create PRs for both repos.

File: controllers/release/adapter_test.go
Comment (davidmogar) [reply controllers/release/adapter_test.go]: What indeed.

File: controllers/release/adapter_test.go
Comment (davidmogar) [reply controllers/release/adapter_test.go:75]: Nothing is carried over between suites.

File: controllers/release/adapter.go
Comment (davidmogar) [reply controllers/release/adapter.go]: Changed to just `managed`.

File: tekton/utils.go
Comment (davidmogar) [reply tekton/utils.go]: This was being changed in the next PR, but I added a constant here for managed Pipelines.

File: tekton/utils/suite_test.go
Comment (davidmogar) [reply tekton/utils/suite_test.go]: Added after rebasing. Removed.

File: tekton/utils/pipeline_run_builder.go
Comment (johnbieren) [inline tekton/utils/pipeline_run_builder.go:233]: This still works if `defaultTimeouts` are not defined in the RSC, right?

File: tekton/utils/pipeline_run_builder.go
Comment (johnbieren) [inline tekton/utils/pipeline_run_builder.go:196]: This part is still untested I think

File: tekton/utils/pipeline.go
Comment (johnbieren) [reply tekton/utils/pipeline.go:57]: Sweet, leaving this open until those PRs are open and merged as this should not be merged until that is the case

File: tekton/utils_test.go
Comment (johnbieren) [inline tekton/utils_test.go:28]: `s/release/managed/`

File: tekton/utils_test.go
Comment (johnbieren) [inline tekton/utils_test.go:50]: Copy paste error here I think

File: tekton/utils/pipeline_run_builder.go
Comment (davidmogar) [reply tekton/utils/pipeline_run_builder.go:233]: Yes, it should as it will set it to nil which was the previous behavior.

File: tekton/utils/pipeline_run_builder.go
Comment (davidmogar) [reply tekton/utils/pipeline_run_builder.go:196]: I copied the test we had before (`It("adds the taskGit pipeline parameters to the PipelineRun object when using a git resolver", func() {`. Not sure what you mean.

Comment (johnbieren) [reply tekton/utils/pipeline_run_builder.go:196]: `adds the taskGit` does not exist in `tekton/utils/pipeline_run_builder_test.go` 
?

Comment (davidmogar) [reply tekton/utils/pipeline_run_builder.go:197]: Maybe it wasn't pushed. It's there now.

File: tekton/utils/suite_test.go
Comment (johnbieren) [reply tekton/utils/suite_test.go:58]: This comment suggests these changes aren't needed and are just from a rebase gone wrong

File: api/v1alpha1/releaseserviceconfig_types.go
Comment (johnbieren) [inline api/v1alpha1/releaseserviceconfig_types.go:38]: shouldn't have a period but that is a nit

**Review by johnbieren — APPROVED** [review]

**Comment by johnbieren** [comment]
Blocked by https://github.com/redhat-appstudio/e2e-tests/pull/1055

**Pull request merged**
